overlord/logstate/manager.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package logstate
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/overlord/state"
+	"github.com/juju/loggo"
+	"github.com/juju/version/v2"
+)
+
+type State interface {
+	DB() *sql.DB
+}
+
+type LogManager struct {
+	state State
+
+	mutex      sync.Mutex
+	statements map[string]*sql.Stmt
+}
+
+func NewManager(s State) *LogManager {
+	mgr := &LogManager{
+		state:      s,
+		statements: make(map[string]*sql.Stmt),
+	}
+	return mgr
+}
+
+// StartUp the LogManager preparing the statements required for appending lines.
+func (m *LogManager) StartUp(ctx context.Context) error {
+	db := m.state.DB()
+	stmt, err := db.PrepareContext(ctx, sqlInsertLogEntry)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	m.statements[sqlInsertLogEntry] = stmt
+
+	return nil
+}
+
+func (m *LogManager) Ensure() error {
+	return nil
+}
+
+func (m *LogManager) Stop() error {
+	for _, stmt := range m.statements {
```

Comment:
```go
  m.mutex.Lock()
  defer m.mutex.Unlock()

  for {}
 
  m.statements = nil
}
```

If you close the manager you should ensure that future calls to its other methods return errors; this one becomes a no-op.